src/test/java/com/puppycrawl/tools/checkstyle/TreeWalkerTest.java

```diff
@@ -253,6 +262,21 @@ public void testRequiredTokenIsEmptyIntArray() throws Exception {
         }
     }
 
+    @Test
+    public void testBehaviourWithZeroChecks() throws Exception {
+        final TreeWalker treeWalker = new TreeWalker();
+        final String errMsg = "There should be no parsing or walking ASTs if there are no checks";
+        mockStatic(TreeWalker.class);
+        when(TreeWalker.parse(any(FileContents.class)))
+                .thenThrow(new IllegalStateException(errMsg));
+        try {
+            treeWalker.processFiltered(temporaryFolder.newFile("file.java"), new ArrayList<>());
+        }
+        catch (IllegalStateException ex) {
+            fail(ex.getMessage());
```

Comment:
'Catch-fail' is not needed. If any exception is thrown outside method, junit will fail as default and give the user the full stack trace.
Catch is only needed to verify exception occurs and the message/trace it gives.

remove try/catch.